cachito/web/models.py

```diff
@@ -893,35 +868,25 @@ def _state_to_json(state):
             latest_state = states[0]
             rv["state_history"] = states
 
-            package_to_deps = {}
-            # This is used to quickly see if a dependency has been added to the root "dependencies"
-            # array in the returned JSON. This uses a set instead of iterating through the
-            # "dependencies" array to avoid an O(n) operation for every dependency.
-            seen_dependencies = set()
+            rv["packages"] = []
             rv["dependencies"] = []
-            for req_dep in self.request_dependencies:
-                dep_json = req_dep.dependency.to_json(
-                    req_dep.replaced_dependency.to_json() if req_dep.replaced_dependency else None,
-                    force_replaces=True,
-                )
-                # Avoid duplicate dependencies in the root "dependencies" array
-                if req_dep.dependency.id not in seen_dependencies:
-                    rv["dependencies"].append(dep_json)
-                    seen_dependencies.add(req_dep.dependency.id)
 
-                package_to_deps.setdefault(req_dep.package.id, []).append(dep_json)
-
-            sort_packages_and_deps_in_place(rv["dependencies"])
-
-            rv["packages"] = [
-                req_package.package.to_json(
-                    dependencies=package_to_deps.get(req_package.package_id, []),
-                    subpath=req_package.subpath,
-                )
-                for req_package in self.request_packages
-            ]
-
-            sort_packages_and_deps_in_place(rv["packages"])
+            bundle_dir = RequestBundleDir(
+                self.id, root=flask.current_app.config["CACHITO_BUNDLES_DIR"]
+            )
+            if bundle_dir.packages_data.exists():
```

Comment:
I will need to load the packages data as well to generate the content manifest (CLOUDBLD-4730), so I was thinking that it could be a good idea to make this reusable. I have two suggestions on how it could be made:

1- Load the package data from the json file right after the object is loaded from the database. This way the data would always be available and its behavior would be similar to having the packages on the database.

2- Have this behavior be done on private function on the class, so it would be easy to just reuse this part of the code. This way, it's easier to implement and we'd have to worry less about corner cases of suggestion 1.

Reply:
I have changes to the `PackagesData` locally, I'll push it later and please review it in advance.

Reply:
I like it, I think it looks a lot better now :)